### ISSUE TITLE
Missing methods erf(x, RoundUp)

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalSpecialFunctions.jl#4]
We're missing methods like `erf(x, RoundUp)`.

cc @SebastianM-C

**Comment by SebastianM-C** [comment]
I am not sure I understand what this part does:
```julia
for f in (:erf, :erfc)

    @eval function($f)(x::BigFloat, r::RoundingMode)
        setrounding(BigFloat, r) do
            ($f)(x)
        end
    end

    @eval ($f)(a::Interval{Float64}) = convert(Interval{Float64}, ($f)(big53(a)))

end
```

Was this not supposed to define the versions of the functions for all rounding modes?

**Comment by dpsanders** [comment]
Yes, but in line 3 you can see that it only does it for `x` that is a `BigFloat`.

**Comment by dpsanders** [comment]
It uses metaprogramming to automate the generation instead of repeating very similar code twice.
But we can equally just repeat the code twice, which may well be clearer!

**Comment by SebastianM-C** [comment]
The metaprogramming part was quite clear, but I am not sure about the `setrounding` part.

Would it be better to use a parametric type instead of `BigFloat` or should we define explicitly what happens?

**Comment by dpsanders** [comment]
As far as I remember / can see, we have correctly rounded functions only for BigFloat.
So the Float64 ones will have to be defined in terms of those.
